site/content/blog/2020-08-28-introducing-the-graphql-add-on-for-zap/index.md

```diff
@@ -0,0 +1,108 @@
+
+If you do not have the schema, no worries at all! Chances are, the endpoint you want to attack has [Introspection](https://graphql.org/learn/introspection/) enabled by default. That will allow us to generate the schema locally by sending a single large introspection query to the endpoint. ZAP will do this automatically. You just have to specify the endpoint URL via the second option.
+
+![The Import From URL Dialog](images/import-from-url.png)
+
+The add-on also adds rules for the ZAP Spider. So, if during a scan you inadvertently come across a GraphQL endpoint, an introspection query will be sent to it automatically. Note that this will only work with recent weeklies or version 2.10.0 when it is released.
+
+You can also import a GraphQL Schema via the API or the CLI.
+
+Regardless of how a Schema is obtained, if everything is valid and correctly specified, it will be ready for the next stage: query generation.
+
+## Query Generation
+
+The imported schema will be parsed and queries will be generated from it.
+
+What exactly do we mean by *generated*? 
+
+Basically, ZAP parses the imported schema and tries to create all possible valid queries from it. Allow me to explain this with the help of an example.
+
+For example: If the imported schema is,
+
+```
+type  Query {
+   polygon (sides: Int, regular: Boolean): Polygon
+}
+
+type  Polygon {
+   perimeter: Float
+   area: Float
+}
+```
+
+Then one possible generated query is,
+
+```
+query {
+  polygon(sides: 1, regular: true) {
+	perimeter
+	area
+  }
+}
+```
+
+This is in fact a [unit test](https://github.com/zaproxy/zap-extensions/blob/3cb399a8e8a3b998e4b90cf8bffba99501ef00f0/addOns/graphql/src/test/java/org/zaproxy/addon/graphql/GraphQlGeneratorUnitTest.java#L72) for the add-on. You can look at the other tests for more examples.
+
+Now, if you go to *Tools -> Options -> GraphQL*, you will find the following settings panel:
+
+![The GraphQL Options Panel](images/graphql-options-panel.png)
+
+These settings allow you to control the output of the add-on's Query Generator. Based on the combination of your chosen values for the above options, you will be able to send unique queries to the endpoint.
+
+For example, by increasing the maximum query depth, you may be able to exhaust the server's resources. Or if you specify arguments using variables, you could perform injection attacks by making use of ZAP's input vectors. You may also be able to bypass authorization and discover IDORs by generating queries with or without certain arguments and fields. The possibilities are endless!
+
+I did say before, that the add-on is still in an early stage. If you face any problems or have any feature requests, feel free to join in the discussion on the [ZAP User Group](https://groups.google.com/d/msg/zaproxy-users/j0Uxg6cUIY8/ghBP26sjBgAJ) or contact me directly. You can also ping us on the irc channel ([#zaproxy on Freenode](https://webchat.freenode.net/#zaproxy)) or raise a GitHub issue for bugs.
```

Comment:
This link is to an unrelated topic/message, was it meant as `https://groups.google.com/group/zaproxy-users`?